lib/delta-sync/delta-file.js

```diff
@@ -80,12 +86,29 @@ export default class DeltaFile {
   async ingest(onFinishCallback) {
     console.log(`Start ingesting file ${this.id} stored at ${this.deltaFilePath}`);
     try {
+      //Couldn't we use data from a variable on this file object? Much faster than writing and immediately reading from the same file.
       const changeSets = await fs.readJson(this.deltaFilePath, {encoding: 'utf-8'});
+      console.log("WHILE INGESTING, this is a changeSets: ", JSON.stringify(changeSets));
       for (let {inserts, deletes} of changeSets) {
-        console.log(`Deleting data in all graphs`);
-        await deleteTriplesInAllGraphs(deletes);
+        //Filter on file triples
+        let partitionedInserts = partition(inserts, change => change.graph.value === UPLOADED_FILES_GRAPH);
```

Comment:
Yes, on the graph. Graph URI can be configured in the config.js. We could consider supplying an array of URI's for better flexibility.
Filtering on `share://` only filters for the physical file metadata. Virtual file metadata is used for downloading.
We could rely on a specific prefix for all virtual file metadata, but that needs to be configured on the producer and consumer both. This means filtering on at least 2 prefixes for the URI's of triples.
This considering, filtering on graph URI sounds less brittle to me.

Reply:
On the graph is a problem anyway, because a graph might contain more data than files alone. So there is a possibility some triples will never be ingested with this approach (since not necessarily attached to a file). Can we try it out with `share://`?
In the background job, you can then perform the check whether the data is complete to proceed with downloading?

Reply:
All right